pkg/asset/manifests/gcp/cloudproviderconfig.go

```diff
@@ -68,6 +79,8 @@ node-tags       = {{$tag}}
 node-instance-prefix = {{.Global.NodeInstancePrefix}}
 external-instance-groups-prefix = {{.Global.ExternalInstanceGroupsPrefix}}
 subnetwork-name = {{.Global.SubnetworkName}}
-{{ if ne .Global.NetworkProjectID "" }}network-project-id = {{.Global.NetworkProjectID}}{{end}}
+{{- if ne .Global.NetworkProjectID "" }}{{"\n"}}network-project-id = {{.Global.NetworkProjectID}}{{ end }}
+{{- if ne .Global.APIEndpoint "" }}{{"\n"}}api-endpoint = {{.Global.APIEndpoint}}{{ end }}
+{{- if ne .Global.ContainerAPIEndpoint "" }}{{"\n"}}container-api-endpoint = {{.Global.ContainerAPIEndpoint}}{{ end }}
```

Comment:
I think we should be able to drop the new lines `{{"\n"}}`. It's been a while since I've played with templates, but I think what's happening is the beginning `{{-` is removing the whitespace. See [the docs here](https://pkg.go.dev/text/template#hdr-Text_and_spaces) which say:

> if an action's left delimiter (by default "{{") is followed immediately by a minus sign and white space, all trailing white space is trimmed from the immediately preceding text. 

So let me suggest this, but I haven't tested it:
```suggestion
{{ if ne .Global.NetworkProjectID "" }}network-project-id = {{.Global.NetworkProjectID}}{{ end }}
{{ if ne .Global.APIEndpoint "" }}api-endpoint = {{.Global.APIEndpoint}}{{ end }}
{{ if ne .Global.ContainerAPIEndpoint "" }}container-api-endpoint = {{.Global.ContainerAPIEndpoint}}{{ end }}
```

Reply:
> It is also possible to remove whitespace after with the { end -} but this didn't do what I needed either.

Did you try `{{- end }}`? I think that's what you want `{{- ` will remove preceding whitespace, ` -}}` removes following whitespace.

Reply:
```go
{{ if ... }}text{{- end }}
\n{{ if ... }}text{{- end }}
```

IIUC the  `{{- end }}` trims up to the content within the if block, but instead the newline comes at the front of the if block so it will still be there (I tried :cry:). While working with helm charts, I saw a common pattern:

```
# The entry carries its newline 
# so if it is not rendered, the newline is not either
{{- if ... }}
\nText
{{- end }}
```

It is basically what Brent currently has but rewritten like below with a less compact format. WDYT?

```go
{{- if ne .Global.NetworkProjectID "" }}
network-project-id = {{.Global.NetworkProjectID}}
{{- end }}
{{- if ne .Global.APIEndpoint "" }}
api-endpoint = {{.Global.APIEndpoint}}
{{- end }}
{{- if ne .Global.ContainerAPIEndpoint "" }}
container-api-endpoint = {{.Global.ContainerAPIEndpoint}}
{{- end }}
```

Reply:
Thanks. Sounds good to me too :D